src/server/api/routers/application.ts

```diff
@@ -37,7 +37,13 @@ export const applicationRouter = createTRPCRouter({
 
   save: protectedProcedure
     .input(applicationSaveSchema)
-    .mutation(({ input, ctx }) => {
-      // TODO: handler body
+    .mutation( async ({ input, ctx }) => {
+      const userId = ctx.session.user.id;
+      const applicationData = input;
+  
+      const savedApplication = await ctx.db.insert(applications).values({...applicationData, userId})
+        .onConflictDoUpdate({target: [applications.id, applications.userId], set: {...applicationData, userId}})
```

Comment:
Sorry I completely forgot about error handling in my last pass through! Could we wrap this (or the entire function body) in a try catch and throw a `TRPCError` with code "INTERNAL_SERVER_ERROR"?